odoo/service/model.py

```diff
@@ -106,7 +107,18 @@ def _(src):
                 # Automatically retry the typical transaction serialization errors
                 if e.pgcode not in PG_CONCURRENCY_ERRORS_TO_RETRY:
                     raise
-                if tries >= MAX_TRIES_ON_CONCURRENCY_FAILURE:
+                if max_tries is None:
+                    # TODO: refactor to reuse code from tr function above
+                    try:
+                        from odoo.http import request
+                        ctx = request.env.context
+                        max_tries = ctx.get("MAX_TRIES_ON_CONCURRENCY_FAILURE")
+                    except Exception:
+                        pass
+                    if max_tries is None:
+                        max_tries = MAX_TRIES_ON_CONCURRENCY_FAILURE
+
+                if tries >= max_tries:
```

Comment:
You should document this so that it's clear that you're trying to make the retry system abort immediately in a very specific case (bus update).

In fact, I would change the context key to something like `abort_retries=True` and have the condition be something like `if tries >= MAX_TRIES_ON_CONCURRENCY_FAILURE or abort_retries:`

Reply:
Using ``MAX_TRIES_ON_CONCURRENCY_FAILURE`` in context looks more unique than ``about_retires``, plus allows other usage, e.g. set it to 100 if someone prefer avoid exit with and exception